frontend/src/components/game/PlayerGameView.tsx

```diff
@@ -182,17 +189,52 @@ function PlayerGameView(props: PlayerGameViewProps) {
    */
   useBeforeunload(() => 'Leaving this page may cause you to lose your current code and data.');
 
+  const getCurrentLanguage = useCallback(() => languageList[currentProblemIndex],
+    [languageList, currentProblemIndex]);
+
+  const setOneCurrentLanguage = (newLanguage: Language) => {
+    setLanguageList(languageList.map((current, index) => {
+      if (index === currentProblemIndex) {
+        return newLanguage;
+      }
+      return current;
+    }));
+  };
+
+  const setOneCurrentCode = (newCode: string) => {
+    setCodeList(codeList.map((current, index) => {
+      if (index === currentProblemIndex) {
+        return newCode;
+      }
+      return current;
+    }));
+  };
+
+  // Returns the most recent submission made for problem of index curr.
+  const getSubmission = (curr: number, playerSubmissions: Submission[]) => {
+    for (let i = playerSubmissions.length - 1; i >= 0; i -= 1) {
+      if (playerSubmissions[i].problemIndex === curr) {
+        return playerSubmissions[i];
+      }
+    }
+
+    return null;
+  };
```

Comment:
nit: maybe make this into a useCallback? Not sure if there's any major advantage (speed maybe) to doing that or not tho, but could be worth looking into the docs about that

Reply:
Actually, maybe this could even be moved into one of the utils files, as I can imagine this being useful on the results and other pages